--- CMakeLists.txt	2017-01-13 15:12:39.400435517 -0600
+++ CMakeLists.txt.new	2017-01-13 15:12:39.444435152 -0600
@@ -39,6 +39,11 @@
   list(APPEND EXTRA_TARGET_LINK_LIBRARIES pcre)
 endif()
 
+CHECK_CXX_COMPILER_FLAG("-Wno-pedantic" COMPILER_HAS_WNO_PEDANTIC)
+if (COMPILER_HAS_WNO_PEDANTIC)
+  add_compile_options(-Wno-pedantic)
+endif()
+
 include_directories(${CMAKE_SOURCE_DIR})
 
 set(RE2_SOURCES
@@ -68,58 +73,6 @@
 
 add_library(re2 ${RE2_SOURCES})
 
-set(TESTING_SOURCES
-    re2/testing/backtrack.cc
-    re2/testing/dump.cc
-    re2/testing/exhaustive_tester.cc
-    re2/testing/null_walker.cc
-    re2/testing/regexp_generator.cc
-    re2/testing/string_generator.cc
-    re2/testing/tester.cc
-    util/pcre.cc
-    )
-
-add_library(testing STATIC ${TESTING_SOURCES})
-
-set(TEST_TARGETS
-    charclass_test
-    compile_test
-    filtered_re2_test
-    mimics_pcre_test
-    parse_test
-    possible_match_test
-    re2_test
-    re2_arg_test
-    regexp_test
-    required_prefix_test
-    search_test
-    set_test
-    simplify_test
-    string_generator_test
-
-    dfa_test
-    exhaustive1_test
-    exhaustive2_test
-    exhaustive3_test
-    exhaustive_test
-    random_test
-    )
-
-set(BENCHMARK_TARGETS
-    regexp_benchmark
-    )
-
-foreach(target ${TEST_TARGETS})
-  add_executable(${target} re2/testing/${target}.cc util/test.cc)
-  target_link_libraries(${target} testing re2 ${EXTRA_TARGET_LINK_LIBRARIES})
-  add_test(NAME ${target} COMMAND ${target})
-endforeach(target)
-
-foreach(target ${BENCHMARK_TARGETS})
-  add_executable(${target} re2/testing/${target}.cc util/benchmark.cc)
-  target_link_libraries(${target} testing re2 ${EXTRA_TARGET_LINK_LIBRARIES})
-endforeach(target)
-
 set(RE2_HEADERS
     re2/filtered_re2.h
     re2/re2.h
